File: src/main/java/com/mark/GZIPUtils.java
package com.mark;

import java.io.ByteArrayInputStream;
import java.io.ByteArrayOutputStream;
import java.util.zip.Inflater;
import org.apache.commons.compress.compressors.gzip.GzipCompressorInputStream;
import org.apache.commons.compress.compressors.gzip.GzipCompressorOutputStream;

public class GZIPUtils {

    public static byte[] gzipBytes(byte[] input) {
        try (ByteArrayOutputStream bout = new ByteArrayOutputStream();
             GzipCompressorOutputStream gzipper = new GzipCompressorOutputStream(bout))
        {

            gzipper.write(input, 0, input.length);
            gzipper.close();

            return bout.toByteArray();
        } catch(Exception ex) {
            return null;
        }
    }

    public static byte[] unzip(byte[] input)  {
        try (ByteArrayInputStream bin = new ByteArrayInputStream(input);
             GzipCompressorInputStream gzipper = new GzipCompressorInputStream(bin))
        {
            ByteArrayOutputStream out = new ByteArrayOutputStream();
            final byte[] buffer = new byte[2048];
            int n = 0;
            while (-1 != (n = gzipper.read(buffer))) {
                out.write(buffer, 0, n);
            }
            bin.close();
            gzipper.close();
            return out.toByteArray();
        } catch(Exception ex) {
            return null;
        }
    }




}

File: src/main/kotlin/com/mark/utils/GZIPUtils.kt
package com.mark.utils

import org.apache.commons.compress.compressors.gzip.GzipCompressorInputStream
import org.apache.commons.compress.compressors.gzip.GzipCompressorOutputStream
import java.io.ByteArrayInputStream
import java.io.ByteArrayOutputStream

object GZIPUtils {
    fun gzipBytes(input: ByteArray, gzip : Boolean) = ByteArrayOutputStream().use { bout ->
        if (!gzip) {
            bout.toByteArray()
        } else {
            GzipCompressorOutputStream(bout).use {
                it.write(input, 0, input.size)
                it.close()
                bout.toByteArray()
            }
        }

    }?: error("Gzip was Null")

    fun unzip(input: ByteArray) = ByteArrayInputStream(input).use { bin ->
        GzipCompressorInputStream(bin).use {
            val out = ByteArrayOutputStream()
            val buffer = ByteArray(2048)
            var n = 0
            while (-1 != it.read(buffer).also { n = it }) {
                out.write(buffer, 0, n)
            }
            bin.close()
            it.close()
            out.toByteArray()
        }
    }?: error("Unzip was Null")

}